packages/backend/src/services/leasing-service/adapters/core-adapter.ts

```diff
@@ -177,7 +179,14 @@ const createNoteOfInterestForInternalParkingSpace = async (params: {
 
     return { ok: true, data: response.data.content }
   } catch (err) {
-    return { ok: false, err }
+    if (
+      err instanceof AxiosError &&
+      err.response?.status === HttpStatusCode.BadRequest &&
+      err.response.data.content.reason === 'Internal check failed' //todo: update to correct format in core
```

Comment:
onecore-core skulle kunna ansvara för att returnera de felkoder som finns också så slipper vi mappa från rena felmeddelandesträngar här (relativt stor risk att de ändras och blir struligt att hitta när det väl händer).

Detta borde isf också gå in i en delad responstyp t.ex `CreateNoteOfInterestErrorCodes`.
Då stärker vi ytterligare kontraktet lite grann mellan core och konsumenter

Reply:
This should be enough in `onecore-types` for now, right? 

```
enum CreditCheckStatus {
  InternalCreditCheckFailed = 'internal-credit-check-failed',
  Unknown = 'unknown'
}
```

Reply:
Yepp